Static GCC build for MinGW32 fails

crosstool-ng version: 1.27.0
Build system: x86_64-linux-gnu
Host system: x86_64-w64-mingw32
GCC version: 12.4.0

Build output
```
[ALL  ]    Links are now set up to build a cross-compiler
[ALL  ]     from x86_64-build_pc-linux-gnu to arm-unknown-linux-gnueabi.
[ALL  ]    checking for exported symbols... objdump: conftest: not a dynamic object
[ALL  ]    yes
[ALL  ]    checking for -rdynamic... objdump: conftest: not a dynamic object
[ALL  ]    no
[ALL  ]    checking for library containing dlopen... (cached) none required
[ALL  ]    checking for -fPIC -shared... no
[ERROR]    configure: error: 
[ALL  ]       Building GCC with plugin support requires a host that supports
[ALL  ]       -fPIC, -shared, -ldl and -rdynamic.
[ERROR]    make[2]: *** [Makefile:4559: configure-gcc] Error 1
[ALL  ]    make[2]: Leaving directory '/home/oliver/ct-build/.build/HOST-x86_64-w64-mingw32/arm-unknown-linux-gnueabi/build/build-cc-gcc-final-build-x86_64-build_pc-linux-gnu'
[ERROR]    make[1]: *** [Makefile:1034: all] Error 2
[ALL  ]    make[1]: Leaving directory '/home/oliver/ct-build/.build/HOST-x86_64-w64-mingw32/arm-unknown-linux-gnueabi/build/build-cc-gcc-final-build-x86_64-build_pc-linux-gnu'
```

Hmm. Disabling gcc plug-ins should sort that but ct-ng should probably stop
that from being selected. Could you attach your .config and build.log

On Sat, 8 Feb 2025, 5:01 am Oliver Old, ***@***.***> wrote:

> crosstool-ng version: 1.27.0
> Build system: x86_64-linux-gnu
> Host system: x86_64-w64-mingw32
> GCC version: 12.4.0
>
> Build output
>
> [ALL  ]    Links are now set up to build a cross-compiler
> [ALL  ]     from x86_64-build_pc-linux-gnu to arm-unknown-linux-gnueabi.
> [ALL  ]    checking for exported symbols... objdump: conftest: not a dynamic object
> [ALL  ]    yes
> [ALL  ]    checking for -rdynamic... objdump: conftest: not a dynamic object
> [ALL  ]    no
> [ALL  ]    checking for library containing dlopen... (cached) none required
> [ALL  ]    checking for -fPIC -shared... no
> [ERROR]    configure: error:
> [ALL  ]       Building GCC with plugin support requires a host that supports
> [ALL  ]       -fPIC, -shared, -ldl and -rdynamic.
> [ERROR]    make[2]: *** [Makefile:4559: configure-gcc] Error 1
> [ALL  ]    make[2]: Leaving directory '/home/oliver/ct-build/.build/HOST-x86_64-w64-mingw32/arm-unknown-linux-gnueabi/build/build-cc-gcc-final-build-x86_64-build_pc-linux-gnu'
> [ERROR]    make[1]: *** [Makefile:1034: all] Error 2
> [ALL  ]    make[1]: Leaving directory '/home/oliver/ct-build/.build/HOST-x86_64-w64-mingw32/arm-unknown-linux-gnueabi/build/build-cc-gcc-final-build-x86_64-build_pc-linux-gnu'
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/crosstool-ng/crosstool-ng/issues/2288>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABGAEY2KHHYMASLDO4RLE32OTKE3AVCNFSM6AAAAABWWI3USCVHI2DSMVQWIX3LMV43ASLTON2WKOZSHAZTQNJRGU4DONI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/config/binutils/binutils.in
+++ b/config/binutils/binutils.in
@@ -132,6 +132,7 @@ config BINUTILS_LINKER_DEFAULT
 config BINUTILS_PLUGINS
     bool
     prompt "Enable support for plugins"
+    depends on !STATIC_TOOLCHAIN
     help
       binutils can be extended through the use of plugins.
       Especially, gold can use the lto-plugin, as installed

